test/modules/generic-mqtt-tester/src/tester.rs

```diff
@@ -127,39 +134,47 @@ impl MessageTester {
         let test_result_coordinator_url = settings.trc_url();
         let reporting_client = TrcClient::new(test_result_coordinator_url);
 
-        let message_handler: Box<dyn MessageHandler + Send> = match settings.test_scenario() {
-            TestScenario::Initiate => {
-                let batch_id = settings
-                    .batch_id()
-                    .ok_or(MessageTesterError::MissingBatchId)?;
-                Box::new(ReportResultMessageHandler::new(
+        let message_handler: Option<Box<dyn MessageHandler + Send>> = match settings.test_scenario()
+        {
+            TestScenario::InitiateAndReceiveRelayed | TestScenario::Receive => {
+                Some(Box::new(ReportResultMessageHandler::new(
                     reporting_client.clone(),
                     tracking_id,
-                    batch_id,
                     &module_name,
-                ))
+                )))
             }
-            TestScenario::Relay => Box::new(RelayingMessageHandler::new(
+            TestScenario::Relay => Some(Box::new(RelayingMessageHandler::new(
                 publish_handle.clone(),
                 relay_topic,
-            )),
+            ))),
+            TestScenario::Initiate => None,
         };
-        let message_channel = MessageChannel::new(message_handler);
+
+        let mut message_channel = None;
+        let mut message_channel_shutdown = None;
+        if let Some(message_handler) = message_handler {
+            let channel = MessageChannel::new(message_handler);
+            message_channel_shutdown = Some(channel.shutdown_handle());
+            message_channel = Some(channel);
+        }
 
         let mut message_initiator = None;
         let mut message_initiator_shutdown = None;
-        if let TestScenario::Initiate = settings.test_scenario() {
-            let initiator = MessageInitiator::new(publish_handle, reporting_client, &settings)?;
+        match settings.test_scenario() {
+            TestScenario::Initiate | TestScenario::InitiateAndReceiveRelayed => {
+                let initiator = MessageInitiator::new(publish_handle, reporting_client, &settings)?;
 
-            message_initiator_shutdown = Some(initiator.shutdown_handle());
-            message_initiator = Some(initiator);
+                message_initiator_shutdown = Some(initiator.shutdown_handle());
+                message_initiator = Some(initiator);
+            }
+            _ => {}
         }
 
         let (poll_client_shutdown_send, poll_client_shutdown_recv) = mpsc::channel::<()>(1);
```

Comment:
Are we ever going to do this more than once? Basically curious if a oneshot will suffice?

Reply:
There was a comment about this on an earlier PR by Vadim. I initially tried oneshot but ran into problems deriving Clone